oscore-cf/src/main/java/org/eclipse/californium/oscore/ByteId.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2018 RISE SICS and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Tobias Andersson (RISE SICS)
+ *    
+ ******************************************************************************/
+package org.eclipse.californium.oscore;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+
+/**
+ * 
+ * @author Tobias Andersson
+ *
+ * Provides byte arrays with identification.
+ *
+ */
+
+public class ByteId {
+
+		private byte[] id;
+
+		public ByteId(byte[] id) {
+			if(id != null){
+				this.id = id;
+			}else{
+				throw new IllegalArgumentException(OSExceptionHandler.TOKEN_NULL);
+			}
+		}
+
+		@Override
+		public int hashCode() {
+			MessageDigest md = null;
+			try {
+				md = MessageDigest.getInstance("MD5");
+				return ByteBuffer.wrap(md.digest(this.id)).getInt();
+			} catch (Exception e) {
+				e.printStackTrace();
+				throw new RuntimeException(e.getMessage());
+			}
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (!(o instanceof ByteId)) {
+				return false;
+			}
+			ByteId other = (ByteId) o;
+			if (other.id.length != this.id.length) {
```

Comment:
`Arrays.equals` should to it.

Reply:
Maybe I should just remove this class. Array.equals should satisfy the need.

Reply:
You use it as key for an hashmap, so a plain byte[] will not work.

Reply:
True.